src/components/Form/TextInput.tsx

```diff
@@ -67,21 +67,65 @@ export class TextInput extends Component<TextInputProps> {
       ...passProps
     } = this.props;
 
-    const ifScreenreader = labelMode === 'screenreader';
-    if (ifScreenreader && !!placeholder) {
-      logger.warn(
-        'Placeholder not used as label is hidden and using label instead',
+    const hideLabel = labelMode === 'hidden';
+
+    return (
+      <HtmlLabel
+        {...labelProps}
+        className={classnames(labelBaseClassName, className, {
+          [disabledClassName]: !!passProps.disabled,
+        })}
+      >
+        {hideLabel ? (
+          <VisuallyHidden>{labelText}</VisuallyHidden>
+        ) : (
+          <Paragraph {...labelTextProps}>{labelText}</Paragraph>
+        )}
+        <HtmlDiv {...inputContainerProps}>
+          <HtmlInput
+            {...passProps}
+            className={classnames(inputBaseClassName, inputClassName)}
+            type="text"
+            placeholder={hideLabel ? labelText : !!placeholder ? labelText : ''}
```

Comment:
Maybe instead of empty string, '', with the last option use `undefined`.

Because when it is now empty string, React seems to add the `placeholder` attribute to input anyway. And reading the code it seems that it is not wanted.

E.g what happened when the last option was happening:
`<input class="fi-text-input_input sc-dnqmqq yydqa" placeholder="" type="text">`

Reply:
Might be worth checking how the placeholder behaves with IE since e.g. hidden boolean attribute was not properly added / removed when the value was changed. This might not be the case / might not matter with this component, but maybe worth checking still?